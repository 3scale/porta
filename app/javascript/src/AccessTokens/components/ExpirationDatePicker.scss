@import '~@patternfly/patternfly/patternfly-addons.css';

.pf-c-calendar-month,
.pf-c-form-control-expiration {
  width: 50%;
}

.pf-c-form-control-expiration {
  margin-right: 1em;
}

button.pf-c-form__group-label-help {
  min-width: auto;
  width: auto;
}
