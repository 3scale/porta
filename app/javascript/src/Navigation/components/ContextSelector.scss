// Adjust styles for a context selector that is within the (dark) header
.header-context-selector {
  --pf-c-context-selector__toggle--BorderTopColor: transparent;
  --pf-c-context-selector__toggle--BorderRightColor: transparent;
  --pf-c-context-selector__toggle--BorderBottomColor: transparent;
  --pf-c-context-selector__toggle--BorderLeftColor: transparent;
  --pf-c-context-selector__toggle--active--BorderBottomColor: transparent;
  --pf-c-context-selector__toggle--expanded--BorderBottomColor: transparent;
  --pf-c-context-selector__toggle--hover--BorderBottomColor: transparent;
  --pf-c-context-selector--Width: initial;
}

.header-context-selector__toggle-text-icon {
  margin-right: var(--pf-global--spacer--sm);
  color: var(--pf-global--active-color--100);
}

.pf-c-context-selector__menu-list {
  overflow-y: auto;
}
