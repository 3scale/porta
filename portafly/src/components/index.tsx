export * from './pages'
export * from './shared'
export * from './AppLayout'
export * from './AppNavExpandable'
export * from './AppNavGroup'
export * from './AppNavItem'
export * from './AppHeaderTools'
export * from './LazyRoute'
export * from './NotFound'
export * from './SwitchWith404'
export * from './util'
export * from './Root'
