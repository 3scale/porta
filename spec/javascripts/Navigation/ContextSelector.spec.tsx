import { mount } from 'enzyme'
import { ContextSelector } from 'Navigation/components/ContextSelector'

import type { Props } from 'Navigation/components/ContextSelector'
import type { Menu } from 'Types'

const audienceLink = '/audience'
const productsLink = '/products'
const backendsLink = '/backends'
const settingsLink = '/settings'

const defaultProps = {
  activeMenu: '' as Menu,
  audienceLink: audienceLink,
  productsLink: productsLink,
  backendsLink: backendsLink,
  settingsLink: settingsLink
}

const mountWrapper = (props: Partial<Props> = {}) => mount(<ContextSelector {...{ ...defaultProps, ...props }} />)

it('should render itself', () => {
  expect(mountWrapper().find(ContextSelector).exists()).toEqual(true)
})

it('should have Dashboard, Audience, Products, Backends and Settings', () => {
  const wrapper = mountWrapper()
  wrapper.find('.pf-c-context-selector__toggle').simulate('click')
  expect(wrapper).toMatchSnapshot()
})

it('should not have Audience if not provided', () => {
  const wrapper = mountWrapper()
  wrapper.setProps({ audienceLink: undefined })
  wrapper.find('.pf-c-context-selector__toggle').simulate('click')
  expect(wrapper).toMatchSnapshot()
})

it('should highlight the selected context', () => {
  const wrapper = mountWrapper()
  wrapper.find('.pf-c-context-selector__toggle').simulate('click')

  wrapper.setProps({ activeMenu: 'buyers' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Audience')

  wrapper.setProps({ activeMenu: 'finance' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Audience')

  wrapper.setProps({ activeMenu: 'cms' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Audience')

  wrapper.setProps({ activeMenu: 'site' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Audience')

  wrapper.setProps({ activeMenu: 'dashboard' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Dashboard')

  wrapper.setProps({ activeMenu: 'personal' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Account Settings')

  wrapper.setProps({ activeMenu: 'account' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Account Settings')

  wrapper.setProps({ activeMenu: 'active_docs' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Account Settings')

  wrapper.setProps({ activeMenu: 'serviceadmin' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Products')

  wrapper.setProps({ activeMenu: 'backend_api' })
  expect(wrapper.find('.current-context')).toHaveLength(1)
  expect(wrapper.find('.current-context').text()).toEqual('Backends')
})

it('should display the current api', () => {
  const wrapper = mountWrapper({ activeMenu: 'serviceadmin' })

  expect(wrapper.find('.fa-cubes').exists()).toBe(true)
  expect(wrapper.find('.pf-c-context-selector__toggle').text()).toContain('Products')

  wrapper.setProps({ activeMenu: 'monitoring' })
  expect(wrapper.find('.fa-cubes').exists()).toBe(true)
  expect(wrapper.find('.pf-c-context-selector__toggle').text()).toContain('Products')

  wrapper.setProps({ activeMenu: 'backend_api' })
  expect(wrapper.find('.fa-cube').exists()).toBe(true)
  expect(wrapper.find('.pf-c-context-selector__toggle').text()).toContain('Backends')
})
