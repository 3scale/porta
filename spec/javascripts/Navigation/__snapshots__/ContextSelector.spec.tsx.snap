// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render itself 1`] = `
<ContextSelector
  menuItems={
    [
      {
        "disabled": true,
        "href": "provider_admin_dashboard_path",
        "icon": "home",
        "title": "Dashboard",
      },
      {
        "disabled": false,
        "href": "audience_link",
        "icon": "bullseye",
        "title": "Audience",
      },
      {
        "disabled": false,
        "href": "admin_services_path",
        "icon": "cubes",
        "title": "Products",
      },
      {
        "disabled": false,
        "href": "provider_admin_backend_apis_path",
        "icon": "cube",
        "title": "Backends",
      },
      {
        "disabled": false,
        "href": "settings_link",
        "icon": "cog",
        "title": "Account Settings",
      },
    ]
  }
  toggle={
    {
      "icon": "home",
      "title": "Dashboard",
    }
  }
>
  <Dropdown
    data-quickstart-id="context-selector"
    dropdownItems={
      [
        <DropdownItem
          href="provider_admin_dashboard_path"
          isDisabled={true}
        >
          <InlineIcon
            icon="home"
          />
          Dashboard
        </DropdownItem>,
        <DropdownItem
          href="audience_link"
          isDisabled={false}
        >
          <InlineIcon
            icon="bullseye"
          />
          Audience
        </DropdownItem>,
        <DropdownItem
          href="admin_services_path"
          isDisabled={false}
        >
          <InlineIcon
            icon="cubes"
          />
          Products
        </DropdownItem>,
        <DropdownItem
          href="provider_admin_backend_apis_path"
          isDisabled={false}
        >
          <InlineIcon
            icon="cube"
          />
          Backends
        </DropdownItem>,
        <DropdownItem
          href="settings_link"
          isDisabled={false}
        >
          <InlineIcon
            icon="cog"
          />
          Account Settings
        </DropdownItem>,
      ]
    }
    isOpen={true}
    isPlain={true}
    ouiaId="context-selector"
    toggle={
      <DropdownToggle
        aria-label="Context selector toggle"
        onToggle={[Function]}
      >
        <span
          className="pf-c-context-selector__toggle-text"
        >
          <InlineIcon
            icon="home"
            toggle={true}
          />
          Dashboard
        </span>
      </DropdownToggle>
    }
  >
    <DropdownWithContext
      autoFocus={true}
      className=""
      data-quickstart-id="context-selector"
      direction="down"
      dropdownItems={
        [
          <DropdownItem
            href="provider_admin_dashboard_path"
            isDisabled={true}
          >
            <InlineIcon
              icon="home"
            />
            Dashboard
          </DropdownItem>,
          <DropdownItem
            href="audience_link"
            isDisabled={false}
          >
            <InlineIcon
              icon="bullseye"
            />
            Audience
          </DropdownItem>,
          <DropdownItem
            href="admin_services_path"
            isDisabled={false}
          >
            <InlineIcon
              icon="cubes"
            />
            Products
          </DropdownItem>,
          <DropdownItem
            href="provider_admin_backend_apis_path"
            isDisabled={false}
          >
            <InlineIcon
              icon="cube"
            />
            Backends
          </DropdownItem>,
          <DropdownItem
            href="settings_link"
            isDisabled={false}
          >
            <InlineIcon
              icon="cog"
            />
            Account Settings
          </DropdownItem>,
        ]
      }
      isFlipEnabled={true}
      isGrouped={false}
      isOpen={true}
      isPlain={true}
      isText={false}
      menuAppendTo="inline"
      onSelect={[Function]}
      position="left"
      removeFindDomNode={false}
      toggle={
        <DropdownToggle
          aria-label="Context selector toggle"
          onToggle={[Function]}
        >
          <span
            className="pf-c-context-selector__toggle-text"
          >
            <InlineIcon
              icon="home"
              toggle={true}
            />
            Dashboard
          </span>
        </DropdownToggle>
      }
      zIndex={9999}
    >
      <div
        className="pf-c-dropdown pf-m-expanded"
        data-ouia-component-id="context-selector"
        data-ouia-component-type="PF4/Dropdown"
        data-ouia-safe={true}
        data-quickstart-id="context-selector"
      >
        <DropdownToggle
          aria-haspopup={true}
          aria-label="Context selector toggle"
          getMenuRef={[Function]}
          id="pf-dropdown-toggle-id-1"
          isOpen={true}
          isPlain={true}
          isText={false}
          key=".0"
          onEnter={[Function]}
          onToggle={[Function]}
          parentRef={
            {
              "current": <div
                class="pf-c-dropdown pf-m-expanded"
                data-ouia-component-id="context-selector"
                data-ouia-component-type="PF4/Dropdown"
                data-ouia-safe="true"
                data-quickstart-id="context-selector"
              >
                <button
                  aria-expanded="true"
                  aria-haspopup="true"
                  aria-label="Context selector toggle"
                  class="pf-c-dropdown__toggle pf-m-plain"
                  data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                  data-ouia-component-type="PF4/DropdownToggle"
                  data-ouia-safe="true"
                  id="pf-dropdown-toggle-id-1"
                  type="button"
                >
                  <span
                    class="pf-c-dropdown__toggle-text"
                  >
                    <span
                      class="pf-c-context-selector__toggle-text"
                    >
                      <span
                        class="pf-c-icon pf-m-inline pf-c-icon__toggle-inline-icon"
                      >
                        <span
                          class="pf-c-icon__content"
                        >
                          <i
                            aria-hidden="true"
                            class="fa fa-home"
                          />
                        </span>
                      </span>
                      Dashboard
                    </span>
                  </span>
                  <span
                    class="pf-c-dropdown__toggle-icon"
                  >
                    <svg
                      aria-hidden="true"
                      fill="currentColor"
                      height="1em"
                      role="img"
                      style="vertical-align: -0.125em;"
                      viewBox="0 0 320 512"
                      width="1em"
                    >
                      <path
                        d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                      />
                    </svg>
                  </span>
                </button>
                <ul
                  aria-labelledby="pf-dropdown-toggle-id-1"
                  class="pf-c-dropdown__menu"
                  role="menu"
                >
                  <li
                    role="none"
                  >
                    <a
                      aria-disabled="true"
                      class="pf-m-disabled pf-c-dropdown__menu-item"
                      data-ouia-component-id="OUIA-Generated-DropdownItem-1"
                      data-ouia-component-type="PF4/DropdownItem"
                      data-ouia-safe="true"
                      href="provider_admin_dashboard_path"
                      role="menuitem"
                      tabindex="-1"
                    >
                      <span
                        class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                      >
                        <span
                          class="pf-c-icon__content"
                        >
                          <i
                            aria-hidden="true"
                            class="fa fa-home"
                          />
                        </span>
                      </span>
                      Dashboard
                    </a>
                  </li>
                  <li
                    role="none"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-dropdown__menu-item"
                      data-ouia-component-id="OUIA-Generated-DropdownItem-2"
                      data-ouia-component-type="PF4/DropdownItem"
                      data-ouia-safe="true"
                      href="audience_link"
                      role="menuitem"
                      tabindex="-1"
                    >
                      <span
                        class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                      >
                        <span
                          class="pf-c-icon__content"
                        >
                          <i
                            aria-hidden="true"
                            class="fa fa-bullseye"
                          />
                        </span>
                      </span>
                      Audience
                    </a>
                  </li>
                  <li
                    role="none"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-dropdown__menu-item"
                      data-ouia-component-id="OUIA-Generated-DropdownItem-3"
                      data-ouia-component-type="PF4/DropdownItem"
                      data-ouia-safe="true"
                      href="admin_services_path"
                      role="menuitem"
                      tabindex="-1"
                    >
                      <span
                        class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                      >
                        <span
                          class="pf-c-icon__content"
                        >
                          <i
                            aria-hidden="true"
                            class="fa fa-cubes"
                          />
                        </span>
                      </span>
                      Products
                    </a>
                  </li>
                  <li
                    role="none"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-dropdown__menu-item"
                      data-ouia-component-id="OUIA-Generated-DropdownItem-4"
                      data-ouia-component-type="PF4/DropdownItem"
                      data-ouia-safe="true"
                      href="provider_admin_backend_apis_path"
                      role="menuitem"
                      tabindex="-1"
                    >
                      <span
                        class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                      >
                        <span
                          class="pf-c-icon__content"
                        >
                          <i
                            aria-hidden="true"
                            class="fa fa-cube"
                          />
                        </span>
                      </span>
                      Backends
                    </a>
                  </li>
                  <li
                    role="none"
                  >
                    <a
                      aria-disabled="false"
                      class="pf-c-dropdown__menu-item"
                      data-ouia-component-id="OUIA-Generated-DropdownItem-5"
                      data-ouia-component-type="PF4/DropdownItem"
                      data-ouia-safe="true"
                      href="settings_link"
                      role="menuitem"
                      tabindex="-1"
                    >
                      <span
                        class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                      >
                        <span
                          class="pf-c-icon__content"
                        >
                          <i
                            aria-hidden="true"
                            class="fa fa-cog"
                          />
                        </span>
                      </span>
                      Account Settings
                    </a>
                  </li>
                </ul>
              </div>,
            }
          }
        >
          <Toggle
            aria-haspopup={true}
            aria-label="Context selector toggle"
            bubbleEvent={false}
            className=""
            data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
            data-ouia-component-type="PF4/DropdownToggle"
            data-ouia-safe={true}
            getMenuRef={[Function]}
            id="pf-dropdown-toggle-id-1"
            isActive={false}
            isDisabled={false}
            isOpen={true}
            isPlain={true}
            isPrimary={false}
            isSplitButton={false}
            isText={false}
            onEnter={[Function]}
            onToggle={[Function]}
            parentRef={
              {
                "current": <div
                  class="pf-c-dropdown pf-m-expanded"
                  data-ouia-component-id="context-selector"
                  data-ouia-component-type="PF4/Dropdown"
                  data-ouia-safe="true"
                  data-quickstart-id="context-selector"
                >
                  <button
                    aria-expanded="true"
                    aria-haspopup="true"
                    aria-label="Context selector toggle"
                    class="pf-c-dropdown__toggle pf-m-plain"
                    data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
                    data-ouia-component-type="PF4/DropdownToggle"
                    data-ouia-safe="true"
                    id="pf-dropdown-toggle-id-1"
                    type="button"
                  >
                    <span
                      class="pf-c-dropdown__toggle-text"
                    >
                      <span
                        class="pf-c-context-selector__toggle-text"
                      >
                        <span
                          class="pf-c-icon pf-m-inline pf-c-icon__toggle-inline-icon"
                        >
                          <span
                            class="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              class="fa fa-home"
                            />
                          </span>
                        </span>
                        Dashboard
                      </span>
                    </span>
                    <span
                      class="pf-c-dropdown__toggle-icon"
                    >
                      <svg
                        aria-hidden="true"
                        fill="currentColor"
                        height="1em"
                        role="img"
                        style="vertical-align: -0.125em;"
                        viewBox="0 0 320 512"
                        width="1em"
                      >
                        <path
                          d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                        />
                      </svg>
                    </span>
                  </button>
                  <ul
                    aria-labelledby="pf-dropdown-toggle-id-1"
                    class="pf-c-dropdown__menu"
                    role="menu"
                  >
                    <li
                      role="none"
                    >
                      <a
                        aria-disabled="true"
                        class="pf-m-disabled pf-c-dropdown__menu-item"
                        data-ouia-component-id="OUIA-Generated-DropdownItem-1"
                        data-ouia-component-type="PF4/DropdownItem"
                        data-ouia-safe="true"
                        href="provider_admin_dashboard_path"
                        role="menuitem"
                        tabindex="-1"
                      >
                        <span
                          class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            class="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              class="fa fa-home"
                            />
                          </span>
                        </span>
                        Dashboard
                      </a>
                    </li>
                    <li
                      role="none"
                    >
                      <a
                        aria-disabled="false"
                        class="pf-c-dropdown__menu-item"
                        data-ouia-component-id="OUIA-Generated-DropdownItem-2"
                        data-ouia-component-type="PF4/DropdownItem"
                        data-ouia-safe="true"
                        href="audience_link"
                        role="menuitem"
                        tabindex="-1"
                      >
                        <span
                          class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            class="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              class="fa fa-bullseye"
                            />
                          </span>
                        </span>
                        Audience
                      </a>
                    </li>
                    <li
                      role="none"
                    >
                      <a
                        aria-disabled="false"
                        class="pf-c-dropdown__menu-item"
                        data-ouia-component-id="OUIA-Generated-DropdownItem-3"
                        data-ouia-component-type="PF4/DropdownItem"
                        data-ouia-safe="true"
                        href="admin_services_path"
                        role="menuitem"
                        tabindex="-1"
                      >
                        <span
                          class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            class="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              class="fa fa-cubes"
                            />
                          </span>
                        </span>
                        Products
                      </a>
                    </li>
                    <li
                      role="none"
                    >
                      <a
                        aria-disabled="false"
                        class="pf-c-dropdown__menu-item"
                        data-ouia-component-id="OUIA-Generated-DropdownItem-4"
                        data-ouia-component-type="PF4/DropdownItem"
                        data-ouia-safe="true"
                        href="provider_admin_backend_apis_path"
                        role="menuitem"
                        tabindex="-1"
                      >
                        <span
                          class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            class="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              class="fa fa-cube"
                            />
                          </span>
                        </span>
                        Backends
                      </a>
                    </li>
                    <li
                      role="none"
                    >
                      <a
                        aria-disabled="false"
                        class="pf-c-dropdown__menu-item"
                        data-ouia-component-id="OUIA-Generated-DropdownItem-5"
                        data-ouia-component-type="PF4/DropdownItem"
                        data-ouia-safe="true"
                        href="settings_link"
                        role="menuitem"
                        tabindex="-1"
                      >
                        <span
                          class="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            class="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              class="fa fa-cog"
                            />
                          </span>
                        </span>
                        Account Settings
                      </a>
                    </li>
                  </ul>
                </div>,
              }
            }
            toggleVariant="default"
          >
            <button
              aria-expanded={true}
              aria-haspopup={true}
              aria-label="Context selector toggle"
              className="pf-c-dropdown__toggle pf-m-plain"
              data-ouia-component-id="OUIA-Generated-DropdownToggle-1"
              data-ouia-component-type="PF4/DropdownToggle"
              data-ouia-safe={true}
              disabled={false}
              id="pf-dropdown-toggle-id-1"
              onClick={[Function]}
              onKeyDown={[Function]}
              type="button"
            >
              <span
                className="pf-c-dropdown__toggle-text"
              >
                <span
                  className="pf-c-context-selector__toggle-text"
                >
                  <InlineIcon
                    icon="home"
                    toggle={true}
                  >
                    <Icon
                      className="pf-c-icon__toggle-inline-icon"
                      isInline={true}
                    >
                      <span
                        className="pf-c-icon pf-m-inline pf-c-icon__toggle-inline-icon"
                      >
                        <span
                          className="pf-c-icon__content"
                        >
                          <i
                            aria-hidden="true"
                            className="fa fa-home"
                          />
                        </span>
                      </span>
                    </Icon>
                  </InlineIcon>
                  Dashboard
                </span>
              </span>
              <span
                className="pf-c-dropdown__toggle-icon"
              >
                <CaretDownIcon
                  color="currentColor"
                  noVerticalAlign={false}
                  size="sm"
                >
                  <svg
                    aria-hidden={true}
                    aria-labelledby={null}
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style={
                      {
                        "verticalAlign": "-0.125em",
                      }
                    }
                    viewBox="0 0 320 512"
                    width="1em"
                  >
                    <path
                      d="M31.3 192h257.3c17.8 0 26.7 21.5 14.1 34.1L174.1 354.8c-7.8 7.8-20.5 7.8-28.3 0L17.2 226.1C4.6 213.5 13.5 192 31.3 192z"
                    />
                  </svg>
                </CaretDownIcon>
              </span>
            </button>
          </Toggle>
        </DropdownToggle>
        <DropdownMenu
          aria-labelledby="pf-dropdown-toggle-id-1"
          autoFocus={false}
          className=""
          component="ul"
          isGrouped={false}
          isOpen={true}
          openedOnEnter={false}
          position="left"
          setMenuComponentRef={[Function]}
        >
          <ul
            aria-labelledby="pf-dropdown-toggle-id-1"
            autoFocus={false}
            className="pf-c-dropdown__menu"
            hidden={false}
            role="menu"
          >
            <DropdownItem
              href="provider_admin_dashboard_path"
              index={0}
              isDisabled={true}
              key=".$Dashboard"
            >
              <InternalDropdownItem
                className=""
                component="a"
                context={
                  {
                    "keyHandler": [Function],
                    "sendRef": [Function],
                  }
                }
                data-ouia-component-id="OUIA-Generated-DropdownItem-1"
                data-ouia-component-type="PF4/DropdownItem"
                data-ouia-safe={true}
                description={null}
                enterTriggersArrowDown={false}
                href="provider_admin_dashboard_path"
                icon={null}
                index={0}
                inoperableEvents={
                  [
                    "onClick",
                    "onKeyPress",
                  ]
                }
                isAriaDisabled={false}
                isDisabled={true}
                isPlainText={false}
                onClick={[Function]}
                role="menuitem"
                styleChildren={true}
                tabIndex={-1}
                tooltipProps={{}}
              >
                <li
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="none"
                >
                  <a
                    aria-disabled={true}
                    className="pf-m-disabled pf-c-dropdown__menu-item"
                    data-ouia-component-id="OUIA-Generated-DropdownItem-1"
                    data-ouia-component-type="PF4/DropdownItem"
                    data-ouia-safe={true}
                    href="provider_admin_dashboard_path"
                    onClick={[Function]}
                    onKeyPress={[Function]}
                    role="menuitem"
                    tabIndex={-1}
                  >
                    <InlineIcon
                      icon="home"
                    >
                      <Icon
                        className="pf-c-icon__inline-icon"
                        isInline={true}
                      >
                        <span
                          className="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            className="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-home"
                            />
                          </span>
                        </span>
                      </Icon>
                    </InlineIcon>
                    Dashboard
                  </a>
                </li>
              </InternalDropdownItem>
            </DropdownItem>
            <DropdownItem
              href="audience_link"
              index={1}
              isDisabled={false}
              key=".$Audience"
            >
              <InternalDropdownItem
                className=""
                component="a"
                context={
                  {
                    "keyHandler": [Function],
                    "sendRef": [Function],
                  }
                }
                data-ouia-component-id="OUIA-Generated-DropdownItem-2"
                data-ouia-component-type="PF4/DropdownItem"
                data-ouia-safe={true}
                description={null}
                enterTriggersArrowDown={false}
                href="audience_link"
                icon={null}
                index={1}
                inoperableEvents={
                  [
                    "onClick",
                    "onKeyPress",
                  ]
                }
                isAriaDisabled={false}
                isDisabled={false}
                isPlainText={false}
                onClick={[Function]}
                role="menuitem"
                styleChildren={true}
                tabIndex={-1}
                tooltipProps={{}}
              >
                <li
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="none"
                >
                  <a
                    aria-disabled={false}
                    className="pf-c-dropdown__menu-item"
                    data-ouia-component-id="OUIA-Generated-DropdownItem-2"
                    data-ouia-component-type="PF4/DropdownItem"
                    data-ouia-safe={true}
                    href="audience_link"
                    role="menuitem"
                    tabIndex={-1}
                  >
                    <InlineIcon
                      icon="bullseye"
                    >
                      <Icon
                        className="pf-c-icon__inline-icon"
                        isInline={true}
                      >
                        <span
                          className="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            className="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-bullseye"
                            />
                          </span>
                        </span>
                      </Icon>
                    </InlineIcon>
                    Audience
                  </a>
                </li>
              </InternalDropdownItem>
            </DropdownItem>
            <DropdownItem
              href="admin_services_path"
              index={2}
              isDisabled={false}
              key=".$Products"
            >
              <InternalDropdownItem
                className=""
                component="a"
                context={
                  {
                    "keyHandler": [Function],
                    "sendRef": [Function],
                  }
                }
                data-ouia-component-id="OUIA-Generated-DropdownItem-3"
                data-ouia-component-type="PF4/DropdownItem"
                data-ouia-safe={true}
                description={null}
                enterTriggersArrowDown={false}
                href="admin_services_path"
                icon={null}
                index={2}
                inoperableEvents={
                  [
                    "onClick",
                    "onKeyPress",
                  ]
                }
                isAriaDisabled={false}
                isDisabled={false}
                isPlainText={false}
                onClick={[Function]}
                role="menuitem"
                styleChildren={true}
                tabIndex={-1}
                tooltipProps={{}}
              >
                <li
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="none"
                >
                  <a
                    aria-disabled={false}
                    className="pf-c-dropdown__menu-item"
                    data-ouia-component-id="OUIA-Generated-DropdownItem-3"
                    data-ouia-component-type="PF4/DropdownItem"
                    data-ouia-safe={true}
                    href="admin_services_path"
                    role="menuitem"
                    tabIndex={-1}
                  >
                    <InlineIcon
                      icon="cubes"
                    >
                      <Icon
                        className="pf-c-icon__inline-icon"
                        isInline={true}
                      >
                        <span
                          className="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            className="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-cubes"
                            />
                          </span>
                        </span>
                      </Icon>
                    </InlineIcon>
                    Products
                  </a>
                </li>
              </InternalDropdownItem>
            </DropdownItem>
            <DropdownItem
              href="provider_admin_backend_apis_path"
              index={3}
              isDisabled={false}
              key=".$Backends"
            >
              <InternalDropdownItem
                className=""
                component="a"
                context={
                  {
                    "keyHandler": [Function],
                    "sendRef": [Function],
                  }
                }
                data-ouia-component-id="OUIA-Generated-DropdownItem-4"
                data-ouia-component-type="PF4/DropdownItem"
                data-ouia-safe={true}
                description={null}
                enterTriggersArrowDown={false}
                href="provider_admin_backend_apis_path"
                icon={null}
                index={3}
                inoperableEvents={
                  [
                    "onClick",
                    "onKeyPress",
                  ]
                }
                isAriaDisabled={false}
                isDisabled={false}
                isPlainText={false}
                onClick={[Function]}
                role="menuitem"
                styleChildren={true}
                tabIndex={-1}
                tooltipProps={{}}
              >
                <li
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="none"
                >
                  <a
                    aria-disabled={false}
                    className="pf-c-dropdown__menu-item"
                    data-ouia-component-id="OUIA-Generated-DropdownItem-4"
                    data-ouia-component-type="PF4/DropdownItem"
                    data-ouia-safe={true}
                    href="provider_admin_backend_apis_path"
                    role="menuitem"
                    tabIndex={-1}
                  >
                    <InlineIcon
                      icon="cube"
                    >
                      <Icon
                        className="pf-c-icon__inline-icon"
                        isInline={true}
                      >
                        <span
                          className="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            className="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-cube"
                            />
                          </span>
                        </span>
                      </Icon>
                    </InlineIcon>
                    Backends
                  </a>
                </li>
              </InternalDropdownItem>
            </DropdownItem>
            <DropdownItem
              href="settings_link"
              index={4}
              isDisabled={false}
              key=".$Account Settings"
            >
              <InternalDropdownItem
                className=""
                component="a"
                context={
                  {
                    "keyHandler": [Function],
                    "sendRef": [Function],
                  }
                }
                data-ouia-component-id="OUIA-Generated-DropdownItem-5"
                data-ouia-component-type="PF4/DropdownItem"
                data-ouia-safe={true}
                description={null}
                enterTriggersArrowDown={false}
                href="settings_link"
                icon={null}
                index={4}
                inoperableEvents={
                  [
                    "onClick",
                    "onKeyPress",
                  ]
                }
                isAriaDisabled={false}
                isDisabled={false}
                isPlainText={false}
                onClick={[Function]}
                role="menuitem"
                styleChildren={true}
                tabIndex={-1}
                tooltipProps={{}}
              >
                <li
                  onClick={[Function]}
                  onKeyDown={[Function]}
                  role="none"
                >
                  <a
                    aria-disabled={false}
                    className="pf-c-dropdown__menu-item"
                    data-ouia-component-id="OUIA-Generated-DropdownItem-5"
                    data-ouia-component-type="PF4/DropdownItem"
                    data-ouia-safe={true}
                    href="settings_link"
                    role="menuitem"
                    tabIndex={-1}
                  >
                    <InlineIcon
                      icon="cog"
                    >
                      <Icon
                        className="pf-c-icon__inline-icon"
                        isInline={true}
                      >
                        <span
                          className="pf-c-icon pf-m-inline pf-c-icon__inline-icon"
                        >
                          <span
                            className="pf-c-icon__content"
                          >
                            <i
                              aria-hidden="true"
                              className="fa fa-cog"
                            />
                          </span>
                        </span>
                      </Icon>
                    </InlineIcon>
                    Account Settings
                  </a>
                </li>
              </InternalDropdownItem>
            </DropdownItem>
          </ul>
        </DropdownMenu>
      </div>
    </DropdownWithContext>
  </Dropdown>
</ContextSelector>
`;
